src/libs/actions/connections/NetSuiteCommands.ts

```diff
@@ -119,7 +139,15 @@ function updateNetSuiteSyncOptionsOnyxData<TSettingName extends keyof Connection
     settingName: TSettingName,
     settingValue: Partial<Connections['netsuite']['options']['config']['syncOptions'][TSettingName]>,
     oldSettingValue: Partial<Connections['netsuite']['options']['config']['syncOptions'][TSettingName]>,
+    modifiedFieldID?: string,
+    pendingAction?: OnyxCommon.PendingAction,
 ) {
+    let syncOptionsOptimisticValue;
+    if (pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
```

Comment:
@shubham1206agra It will set the `DELETE` action only when we click on the `Remove` button. In other cases `UPDATE` action is send

Reply:
Maybe I should give some context in the PR description. Sorry!

My approach with Lists/Segments changed the current flow a bit.
We want to show correct errors (or offline state) for possible actions:
- add list
- remove list
- edit list

We need to do this because the flow is different for some actions:
- ADD - when there is an error while adding a list, remove the record (return to the previous state) after clicking ❌ 
- DELETE, UPDATE - when there is an error while removing/editing a list, return to the old value and show the error

That's why we need to update the `updateNetSuiteCustomLists` function. It takes `modifiedFieldID` and `pendingActions` params and sets pendingAction for the passed fieldID.

I know we update the entire list with the `UpdateNetSuiteCustomLists` command but we only change one field at the moment. That's why I thought it's reasonable to differentiate between pendingActions of the performed steps.

If you see some other solution which you think could work better please share it here 🙏 
Also, any modifications to my approach are appreciated!

Reply:
@shubham1206agra kind bump :)

Reply:
@shubham1206agra We discussed it in [this](https://swmansion.slack.com/archives/C036QM0SLJK/p1722856466635369?thread_ts=1722855119.292609&cid=C036QM0SLJK) Slack thread and it is working that way in the entire app but this is the first place in integrations where we implement it

Reply:
There is a solution for it :)

https://github.com/Expensify/App/pull/46667/commits/2b1acadc1f9ae1cc7dc9f37efd1d25c2a63db381

Reply:
Thanks for noticing! It should be fixed now :)